### ISSUE TITLE
Plan next release

### ISSUE THREAD

**Issue by freshreplicant** [Tuxemon/Tuxemon#1928]
**Is your feature request related to a problem? Please describe.**
It has been some time since the last release (v0.4.34).

**Describe the solution you'd like**
Nightly builds are available but since there has been many changes since August 2022, it might be time to plan for another release. 

**Describe alternatives you've considered**
We could also set some milestones to hit before the next release and then have a period where we test all the changes since last release and mop up any bugs that might have cropped up.

**Additional context**
It may be necessary to revise the existing milestones as part of this.

**Comment by DavidLiang1129** [comment]
I agree, it's a good idea and I'm looking forward to the next release. Also, do you think we should fix #1889 before then in order to be ready for Tuxemon for other operating systems?

**Comment by JaskRendix** [comment]
I don't know honestly. I'm against it.

**Comment by DavidLiang1129** [comment]
> I don't know honestly. I'm against it.

It's a little frustrating, so I wonder why?

**Comment by kerizane** [comment]
Good idea! I will create a branch for the release candidate, and people can spend a week testing it to make sure everything is stable. Any bugs found we can decide how important a fix is.

Normal development can happen against the Dev branch.

I doubt #1889 will happen soon, @NeonFlames what do you think?

Any objections or suggestions to my approach?

Shall I create the branch now, or is there anything else important we want to fix first?

**Comment by DavidLiang1129** [comment]
> Good idea! I will create a branch for the release candidate, and people can spend a week testing it to make sure everything is stable. Any bugs found we can decide how important a fix is.
> 
> Normal development can happen against the Dev branch.
> 
> I doubt #1889 will happen soon, @NeonFlames what do you think?
> 
> Any objections or suggestions to my approach?
> 
> Shall I create the branch now, or is there anything else important we want to fix first?

I agree, but I find that no one seems to notice #1910...

**Comment by freshreplicant** [comment]
> I don't know honestly. I'm against it.

Since you're making the vast majority of the commits at the moment, I would be interested in knowing why you feel that way? Do you have a different vision for when a release should happen?

If you do, maybe we can take the alternative approach offered in my initial post and rework the milestones and arrive at a proper goal for when we (you included) would feel the next release will be ready.

**Comment by JaskRendix** [comment]
> Since you're making the vast majority of the commits at the moment.

Since August 2022. I wasted a lot of time, but it helped me to become more skilled in coding.

> Do you have a different vision for when a release should happen?

Look, the problem isn't the release, but after the release. **v0.4.34** is unplayable on both scenarios because of bugs:
- Spyder crashes if Eyenemy uses Amnesia in the fight in Route 2 (Billie fight) (this https://github.com/Tuxemon/Tuxemon/pull/1338, fixed since September 2022).
- Xero crashes as soon as a player read a sign (this https://github.com/Tuxemon/Tuxemon/pull/1256, fixed since late August).

Do you want to know how many times one of these bugs has been reported?
Xero = 0, Spyder = 1 (late January 2023, shortly after I opened this #1615, but then closed because is pointless).

> If you do, maybe we can take the alternative approach offered in my initial post and rework the milestones and arrive at a proper goal for when we (you included) would feel the next release will be ready.

I agree with you that milestones need to be purged and a lot of issues need to be closed.

Now that you make me think about it, the sooner we get rid of **v0.4.34**, the better.

> Shall I create the branch now, or is there anything else important we want to fix first?

@kerizane if #1937 fixes the issues and #1935 too, then we can create the branch for me.
@freshreplicant let me know if the two PRs fix the issues for you too (especially the one about the crash).

**Comment by kerizane** [comment]
This should be an easy fix https://github.com/Tuxemon/Tuxemon/issues/1910 I'd like to get it into the release. Anything else?

**Comment by freshreplicant** [comment]
> Since August 2022. I wasted a lot of time, but it helped me to become more skilled in coding.

You have been doing some great work and introduced a lot of cool features and Tuxemon has progressed thanks to that work.

> Look, the problem isn't the release, but after the release. **v0.4.34** is unplayable on both scenarios because of bugs:
> 
>     * Spyder crashes if Eyenemy uses Amnesia in the fight in Route 2 (Billie fight) (this [fix Amnesia technique #1338](https://github.com/Tuxemon/Tuxemon/pull/1338), fixed since September 2022).
> 
>     * Xero crashes as soon as a player read a sign (this [Xero scenario fixes: dialog and dialog_chain #1256](https://github.com/Tuxemon/Tuxemon/pull/1256), fixed since late August).
> 
> Do you want to know how many times one of these bugs has been reported? Xero = 0, Spyder = 1 (late January 2023, shortly after I opened this #1615, but then closed because is pointless).
> 
> I agree with you that milestones need to be purged and a lot of issues need to be closed.
> 
> Now that you make me think about it, the sooner we get rid of **v0.4.34**, the better.
> 
> > Shall I create the branch now, or is there anything else important we want to fix first?
> 
> @kerizane if #1937 fixes the issues and #1935 too, then we can create the branch for me. @freshreplicant let me know if the two PRs fix the issues for you too (especially the one about the crash).

So you would like to aim for more stability and bug squashing before the next release? I think that's a good plan, because I have been running into quite a few bugs just testing today, most of which ended up with crashes. Tuxemon has always been quite buggy to be honest, so it would be a good idea to have a decent period of intensive testing to make sure everything is ironed out. It would let your new features shine more.

The nightly Flatpaks make the latest version more accessible for less technical people (at least those using Linux), so that might help.

I'll have a look at testing those fixes and see if I can reproduce the other crashes I was having. I was playing the Flatpak version so I had no console output sadly.

How should we go about cleaning up the Milestones? I've been going through any issues I think might have been resolved and marking them with 'Possible Close', so maybe of them can also be closed.

**Comment by DavidLiang1129** [comment]
I can't wait to test the new version of Tuxemon! I will also invite my friends to test Tuxemon! (If they want)

**Comment by DavidLiang1129** [comment]
To be honest, Tuxemon has really improved a lot, I think it is more and more a "mature game", although to be honest it has been a little bugful recently, but this does not affect my enthusiasm for Tuxemon! I'll help you find bugs!

**Comment by DavidLiang1129** [comment]
> > Since you're making the vast majority of the commits at the moment.
> 
> Since August 2022. I wasted a lot of time, but it helped me to become more skilled in coding.
> 
> > Do you have a different vision for when a release should happen?
> 
> Look, the problem isn't the release, but after the release. **v0.4.34** is unplayable on both scenarios because of bugs:
> - Spyder crashes if Eyenemy uses Amnesia in the fight in Route 2 (Billie fight) (this https://github.com/Tuxemon/Tuxemon/pull/1338, fixed since September 2022).
> - Xero crashes as soon as a player read a sign (this https://github.com/Tuxemon/Tuxemon/pull/1256, fixed since late August).
> 
> Do you want to know how many times one of these bugs has been reported?
> Xero = 0, Spyder = 1 (late January 2023, shortly after I opened this #1615, but then closed because is pointless).
> 
> > If you do, maybe we can take the alternative approach offered in my initial post and rework the milestones and arrive at a proper goal for when we (you included) would feel the next release will be ready.
> 
> I agree with you that milestones need to be purged and a lot of issues need to be closed.
> 
> Now that you make me think about it, the sooner we get rid of **v0.4.34**, the better.
> 
> > Shall I create the branch now, or is there anything else important we want to fix first?
> 
> @kerizane if #1937 fixes the issues and #1935 too, then we can create the branch for me.
> @freshreplicant let me know if the two PRs fix the issues for you too (especially the one about the crash).

I think the reason why no one reports bugs is that Tuxemon is not known yet. I have a way to raise awareness, but I don't know if it's possible... We can attract some Pokemon Hacker or Pokemon fan game writer (I was a Pokemon fan game writer before) After all, there are people like me who just want to write their own story in the framework of Pokemon games, and they have no choice but to use Pokemon, Now we're going to give them the good news that they have a choice, and it's Tuxemon

**Comment by NeonFlames** [comment]
> Good idea! I will create a branch for the release candidate, and people can spend a week testing it to make sure everything is stable. Any bugs found we can decide how important a fix is.
> 
> Normal development can happen against the Dev branch.
> 
> I doubt #1889 will happen soon, @NeonFlames what do you think?
> 
> Any objections or suggestions to my approach?
> 
> Shall I create the branch now, or is there anything else important we want to fix first?

~~First off sorry about the late response, I've been quite busy.~~ On the topic of making a release personally I think that one can be made after fixing the config issue and having a proper play test. However, seeing as the game has evolved quite a bit since the last release just making a new release after fixing the config issue would be fine as well.

**Comment by NeonFlames** [comment]
> Now that you make me think about it, the sooner we get rid of **v0.4.34**, the better.

I would have to agree with this since a lot of game breaking issues have been fixed.

> I doubt https://github.com/Tuxemon/Tuxemon/issues/1889 will happen soon, @NeonFlames what do you think?

I agree with that since fixing the build scripts is low priority due to the fact that one can simply run the source code directly.

**Comment by JaskRendix** [comment]
> I think that's a good plan, because I have been running into quite a few bugs just testing today, most of which ended up with crashes.

I'm used to test what I code, it wasn't happening on my system.

> How should we go about cleaning up the Milestones? I've been going through any issues I think might have been resolved and marking them with 'Possible Close', so maybe of them can also be closed.

@Sanglorian can help with this

---

regarding https://github.com/Tuxemon/Tuxemon/milestone/3

I think you can move this in 0.6:
- #11 (it exists the daycare, but we lack a couple of things)
- #512 (almost done, but we need to refactor statuses into an independent class)

I think:
- #693
- #8
- #4
- #3
- #452
- #489
- #560
- #241

can be closed.

@kerizane what about these?
- #299
- #749

---

regarding https://github.com/Tuxemon/Tuxemon/milestone/7

I think:
- #124
- #249
- ~#375 (was he talking about the cinema mode? if yes, then it's done)~
- #421 
- #574

can be closed

I'll remove these:
- #303 (it looks like a "never-ending process", it's more a project than a step necessary to release);
- #508 (I don't know if it can be a step that needs to be meet, a project? a simple milestone called "feature" in the most generic way?)

**Comment by kerizane** [comment]
#375 the idea for this is that we want to be able to control the camera independently from the player. Is this what the current "cinema mode" does?

**Comment by JaskRendix** [comment]
> https://github.com/Tuxemon/Tuxemon/issues/375 the idea for this is that we want to be able to control the camera independently from the player. Is this what the current "cinema mode" does?

It doesn't, I remove it

**Comment by freshreplicant** [comment]
> > I think that's a good plan, because I have been running into quite a few bugs just testing today, most of which ended up with crashes.
> 
> I'm used to test what I code, it wasn't happening on my system.

Just to clarify, I didn't mean to imply that your code specifically was causing crashes. I trust that you're testing before your PRs. It still wouldn't hurt to test more widely, on different systems and releases, as sometimes bugs are weird like that. I was running into different bugs when pulling straight from git than I was when I was playing the nightly Flatpak build for example.

So I agree with the idea of having some intensive bug testing before the next release. It may also require a short feature freeze maybe - at least that's what I've seen other projects do, but I don't really have any experience as a maintainer.

**Comment by JaskRendix** [comment]
@freshreplicant no problem

@kerizane
```
new conditions since 0.34:

- battle_is
- has_bag
- has_box
- has_party_breeder
- has_sprite
- has_tech
- money_us
- once
- party_infected
- tuxepedia

new actions since 0.34:

add_collision
add_contacts
add_tech
battles_print
breeding
change_bg
clear_kennel
clear_tuxepedia
get_party_monster
kennel_print
money_print
npc_plague
pathfind_to_player
player_sprite
quarantine
random_battle
random_item
random_monster
remove_collision
remove_contacts
set_battle
set_code
set_kennel_visible
set_kennel
set_money
set_monster_plague
set_player_template
set_random_variable
set_tuxepedia
trading
transfer_money
tuxepedia_print
wild_encounter
```

**Comment by DavidLiang1129** [comment]
I'm curious, is a new version coming soon?

**Comment by bitcraft** [comment]
> So you would like to aim for more stability and bug squashing before the next release? I think that's a good plan, because I have been running into quite a few bugs just testing today, most of which ended up with crashes.

I don't think there ever has been a completely bug-free release of Tuxemon.  Its helpful to pick a target and make sure all targets are reached before making a release.  When I was more active and doing releases, I would just pick some arbitrary point and make sure the single player campaign is *generally* playable.  For people who never build or use git, they rely on just downloading the zip (or flatpak) and a buggy update is better than no update at all.

I haven't looked at milestones in a while.  If they should be changed thats fine.  But to me, I'm not like a huge fanatic about semvar and if we have version `0.5039` before version `0.6` then thats fine.  A lot of time can be wasted trying to put things onto arbitrary cutoffs and version numbers.... it's not so important to follow it, especially for a game.

Based on the set of changes since .34, its probably best to just fix up `development` branch so it runs and you can do the basic parts of the game.  A good benchmark to me is:

- does not crash when launching
- able to run the default campaign
- save and load
- battle and capture
- chat, move around, no horrible movement bugs
- able to progress in campaign for a few map screens....does not have to play until the end

If those requirements are met, its a good time to make a release, imo.

**Comment by freshreplicant** [comment]
Hey @bitcraft, nice to see you again!

All good points, I agree with you. 

I think @JaskRendix  actually knocked out a few crashes recently, I encountered a crash in the Cathedral healing centre and that was fixed. I did run into a crash on a wild encounter on the first route but couldn't reproduce it.

Just having a few of us play might turn up one or more two of bugs that can be ironed out so that we comfortably meet the criteria you set out in your post.

**Comment by DavidLiang1129** [comment]
I would like to make one suggestion: 
It's been almost a year since the release of 0.4.34, and in this year Tuxemon has brought countless exciting new features (including but not limited to mini-games, "player" options in menus, Tuxepedia, NuPhone, better "Settings," new backgrounds before games start, etc.) and many fewer bugs. So I think it's almost time to release a new version of RC (this is just my opinion, if anyone has other ideas, welcome to discuss, thanks!).

**Comment by ultidonki** [comment]
I've recently played through the entire Spyder campaign, from beginning to end, without cheating or using command-line actions to progress through the game, and all the issues I've encountered have been fixed. :smile:

Commit 03dc6c82 is the latest commit that fixed some of the issues I've raised - I'm wondering if we should test that one a bit more, and then release that one as the next flatpak version? After that, it should be easier to keep a more regular flatpak release going, as it can just be released every time a few bugs or new features are added. 

If I have time in the next few days I'll test it for an hour or two, just the first village or two in Spyder, and I'll report how it goes. (or someone else can)

**Comment by Sanglorian** [comment]
> Commit [03dc6c8](https://github.com/Tuxemon/Tuxemon/commit/03dc6c82db53c0c4def318f14ff18f6e6bcc92bb) is the latest commit that fixed some of the issues I've raised - I'm wondering if we should test that one a bit more, and then release that one as the next flatpak version? After that, it should be easier to keep a more regular flatpak release going, as it can just be released every time a few bugs or new features are added.

Thank you for playing through the campaign and catching so many bugs and missing elements. 

Love the idea of getting the flatpak updated again. I'd love if we could get a Windows executable version at some point soon too.

**Comment by JaskRendix** [comment]
@ultidonki did you get a chance to play through the expanded plot? I'm wondering if it's worth keeping or if it needs to be trashed since it's not really Spyder anymore. You were tagged in the conversation, let me know what you think.

**Comment by ultidonki** [comment]
> @ultidonki did you get a chance to play through the expanded plot? I'm wondering if it's worth keeping or if it needs to be trashed since it's not really Spyder anymore. You were tagged in the conversation, let me know what you think.

Sorry, I've seen a couple of the videos, but I took a break from Tuxemon and only watched 2 or 3. It looks like they've already been merged in though. I'll look out for them the next time I play through from the beginning :smile:

**Comment by Davidliang11292** [comment]
Long time no see, can anyone tell me when the new version will be released? The development branch hasn't been updated in a long time, thanks

**Comment by wiki-me** [comment]
There are no open issues in the 0.5 [milestone](https://github.com/Tuxemon/Tuxemon/milestone/3), doesn't that mean a release should be made? (plus its been more then 3 years of development).

Another point worth mentioning i think. [flathub](https://flathub.org/en/apps/org.tuxemon.Tuxemon) writes as one of the reason for the software being potentially unsafe is "Uses an end-of-life runtime
The runtime used by this app is no longer receiving security updates".
